includes/multi-currency/Compatibility/WooCommerceSubscriptions.php

```diff
@@ -168,9 +168,14 @@ public function override_selected_currency( $return ) {
 			}
 		}
 
+		// The running_override_selected_currency_filters property is used to avoid an infinite loop
+		// that can occur on the product page when `get_subscription()` is used.
 		$switch_id = $this->get_subscription_switch_id_from_superglobal();
 		if ( $switch_id ) {
-			return get_post_meta( $switch_id, '_order_currency', true );
+			$this->running_override_selected_currency_filters = true;
```

Comment:
Using new `$this->running_override_selected_currency_filters` introduced in #5404.

Reply:
I wonder if there's a clearer name for that class variable, to make it clear what it does - e.g. something like `inhibit_override_selected_currency` or `prevent_override_selected_currency_infinite_loop`.

Reply:
The current naming makes sense to me, although I might suggest adding the `is_` prefix to indicate it is a bool ([along the lines of this article](https://michaelzanggl.com/articles/tips-on-naming-boolean-variables/)).

In my mind, the method's "running" state is what the variable holds, whereas the statement within the method handles the "inhibiting".

FWIW I didn't name this variable; it was introduced in #5404.